docs/resources/ui-building-blocks/components/user-defined-components/component-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+about **states and state management, [refer here](#)**
+:::
+
+**Component state** refers to the information that a component tracks about its current condition or
+the data it manages internally. This can include data such as whether a button is enabled, the value
+of a slider, or the entries in a dynamically updated list. Component state variables are only
+accessible within the given component's scope.
+
+This type of variable is particularly useful for storing data that affects how the component behaves
+or appears, such as toggling UI elements, keeping track of user choices within the component, or
+caching data pertinent to the component's functionality.
+
+For example,
+
+- In a custom drop-down menu component, you might use a component state variable to keep track of
+  which item is currently selected.
+- In a toggle switch component, you could use a component state variable to store the on/off
+  state based on user interaction.
+  This approach ensures that the state of the component is maintained as it interacts with the user
+  or other parts of the application.
+
+When the value of a component state variable is changed, the component can be re-rendered with the
+updated values, displaying the latest state of the component with these updates.
+
+### Creating a Component State
+
+To create a new **Component State variable** in your component, follow the steps:
+
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/nEmCDqupF7YHUTi4hKvW?embed&show_copy_link=true"
+        title="Create Component State"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+While creating a Component State, the following properties are included:
+
+- **Is List:** This property determines whether the variable can hold multiple values of the same
+  data type (like a list or array) or just a single value.
+
+- **Initial Field Value:** This property sets the default value for the variable when it is first
+  created. It's like setting the starting point or the value that the variable begins with before
+  anything else happens.
+
+- **Nullable:** This property determines whether the variable can have a null value. When "*
+  *Nullable**" is set to true, it means the variable can be empty or have a null value. This is
+  useful when dealing with optional data or scenarios where the absence of a value is valid.
+
+Now, let's apply these concepts to the `isFavourite` variable in the context of the above example:
```

Comment:
I think we can cover this info in the Arcade itself.

Reply:
This is an explanation of the example so I would say it should stay outside of Arcade.